apps/s_client.c

```diff
@@ -3553,4 +3556,70 @@ static char *base64encode (const void *buf, size_t len)
     return out;
 }
 
+/*
+ * Host dNS Name verifier: used for checking that the hostname is in dNS format 
+ * before setting it as SNI
+ */
+static int is_dNS_name(const char *host)
+{
+    const int MAX_LABEL_LENGTH = 63;
+
+    int i;
+    int isdnsname = 0;
+    int length = strlen(host);
+    int label_length = 0;
+    int all_numeric = 1;
```

Comment:
`MAX_LABEL_LENGTH`, `i`, `length` and `label_length` all seem to be more appropriately declared as `size_t`.

Reply:
In `cn2dnsid` the original input is ASN.1 string, and the API for converting those to UTF8 uses integer lengths, so `size_t` was not an option there.  Here, the hostname comes from the command-line, so `size_t` is possible, but does not add any measure of security, I am not aware of any OS where one can execute a program with a 2GB command-line argument...  Still this code could `size_t` just to "unask" the question.

Reply:
I'll change just to make it consistent with the "contents" of the variables

Reply:
Right - my point wasn't really about security. More about using an appropriate and consistent type.

Reply:
Fixed